packages/cosmic-swingset/tools/test-kit.js

```diff
@@ -365,53 +398,90 @@ export const makeCosmicSwingsetTestKit = async (
       },
     });
   };
+
   /**
    * @param {string} fnText must evaluate to a function that will be invoked in
    *   a core eval compartment with a "powers" argument as attenuated by
    *   `jsonPermits` (with no attenuation by default).
    * @param {string} [jsonPermits] must deserialize into a BootstrapManifestPermit
-   * @param {InboundQueue} [queue]
    */
-  const pushCoreEval = (
-    fnText,
-    jsonPermits = 'true',
-    queue = highPriorityQueue,
-  ) => {
+  const pushCoreEval = (fnText, jsonPermits = 'true') => {
     // Fail noisily if fnText does not evaluate to a function.
     // This must be refactored if there is ever a need for such input.
     const fn = new Compartment().evaluate(fnText);
     typeof fn === 'function' || Fail`text must evaluate to a function`;
-    /** @type {import('@agoric/vats/src/core/lib-boot.js').BootstrapManifestPermit} */
-    // eslint-disable-next-line no-unused-vars
-    const permit = JSON.parse(jsonPermits);
     /** @type {import('@agoric/cosmic-proto/swingset/swingset.js').CoreEvalSDKType} */
-    const coreEvalDesc = {
-      json_permits: jsonPermits,
-      js_code: fnText,
-    };
-    const action = {
-      type: QueuedActionType.CORE_EVAL,
-      evals: [coreEvalDesc],
-    };
-    pushQueueRecord(action, queue);
+    const coreEvalDesc = { json_permits: jsonPermits, js_code: fnText };
+    return evaluateProposal({ bundles: [], evals: [coreEvalDesc] });
   };
 
+  /**
+   * @param {Awaited<ReturnType<import('./test-proposal-utils.ts')['buildProposal']>>} proposal
+   */
+  const evaluateProposal = async ({ bundles: proposalBundles, evals }) => {
+    await Promise.resolve();
+
+    for (const bundle of proposalBundles) await installBundle(bundle);
+    pushQueueRecord(
+      { evals, type: QueuedActionType.CORE_EVAL },
+      highPriorityQueue,
+    );
+
+    return runUntilQueuesEmpty();
+  };
+
+  /**
+   * @param {EndoZipBase64Bundle} bundle
+   */
+  const installBundle = bundle => {
+    pushQueueRecord(
+      { bundle: JSON.stringify(bundle), type: QueuedActionType.INSTALL_BUNDLE },
+      highPriorityQueue,
+    );
+    return runNextBlock();
+  };
+
+  const runUntilQueuesEmpty = async () => {
+    await Promise.resolve();
+
+    let stats = controller.getStats();
+
+    while (
+      actionQueue.size() +
+        highPriorityQueue.size() +
+        stats.acceptanceQueueLength +
+        stats.runQueueLength >
+      0
+    ) {
+      await runNextBlock();
+      stats = controller.getStats();
+    }
```

Comment:
Readability suggestion:
```suggestion
    const queuesEmpty = () => {
      const stats = controller.getStats();
      if (highPriorityQueue.size()) return false;
      if (actionQueue.size()) return false;
      if (stats.acceptanceQueueLength) return false;
      if (stats.runQueueLength) return false;
      return true;
    };
    
    while (!queuesEmpty()) await runNextBlock();
```